bundles/org.openhab.binding.digitalstrom/src/main/java/org/openhab/binding/digitalstrom/internal/discovery/DiscoveryServiceManager.java

```diff
@@ -194,20 +198,25 @@ public void onDeviceRemoved(GeneralDeviceInformation device) {
 
     @Override
     public void onDeviceAdded(GeneralDeviceInformation device) {
-        if (device instanceof Device) {
-            String id = ((Device) device).getHWinfo().substring(0, 2);
-            if (((Device) device).isSensorDevice()) {
-                id = ((Device) device).getHWinfo();
-            }
-            if (discoveryServices.get(id) != null) {
-                ((DeviceDiscoveryService) discoveryServices.get(id)).onDeviceAdded(device);
+        try {
+            if (device instanceof Device) {
+                String id = ((Device) device).getHWinfo().substring(0, 2);
+                if (((Device) device).isSensorDevice()) {
+                    id = ((Device) device).getHWinfo();
+                }
+                if (discoveryServices.get(id) != null) {
+                    ((DeviceDiscoveryService) discoveryServices.get(id)).onDeviceAdded(device);
+                }
             }
-        }
-        if (device instanceof Circuit) {
-            if (discoveryServices.get(DsDeviceThingTypeProvider.SupportedThingTypes.circuit.toString()) != null) {
-                ((DeviceDiscoveryService) discoveryServices
-                        .get(DsDeviceThingTypeProvider.SupportedThingTypes.circuit.toString())).onDeviceAdded(device);
+            if (device instanceof Circuit) {
+                if (discoveryServices.get(DsDeviceThingTypeProvider.SupportedThingTypes.circuit.toString()) != null) {
+                    ((DeviceDiscoveryService) discoveryServices
+                            .get(DsDeviceThingTypeProvider.SupportedThingTypes.circuit.toString()))
+                                    .onDeviceAdded(device);
+                }
             }
+        } catch (Exception ex) {
+            logger.error("Unable to add devices {}", device);
```

Comment:
The error logging level should be reserved for issues catastrophic enough to threaten the operation of openHab itself. An error in a binding shouldn't qualify for this. Please read the [openHab coding guidelines](https://www.openhab.org/docs/developer/guidelines.html#f-logging) for expected use of logging in bindings.

I suggest logging warn instead. Also please try to catch specific exceptions or at the very least RuntimeException. In addition, you should always log the exception or at the very least the exception message to make debugging easier.
```suggestion
        } catch (RuntimeException ex) {
            logger.warn("Unable to add devices {}: {}", device, ex.getMessage());
```

Reply:
I guess I wrote this a bit too fast and forgot to forward the exception.

Agree for le level. No problem for me. I will put the complete stacktrace instead of just the message if it's ok for you ?